rs/rosetta-api/icp/tests/system_tests/common/governance_client.rs

```diff
@@ -0,0 +1,107 @@
+    pb::v1::{
+        manage_neuron_response, manage_neuron_response::MakeProposalResponse, MakeProposalRequest,
+        ManageNeuronResponse, ProposalInfo,
+    },
+    proposal_submission_helpers::create_make_proposal_payload,
+};
+
+pub struct GovernanceClient {
+    agent: Agent,
+    governance_principal: Principal,
+}
+
+impl GovernanceClient {
+    pub fn new(agent: Agent, governance_principal: Principal) -> GovernanceClient {
+        GovernanceClient {
+            agent,
+            governance_principal,
+        }
+    }
+
+    pub async fn make_proposal(
+        &self,
+        neuron_id: NeuronId,
+        neuron_controller: PrincipalId,
+        proposal: &MakeProposalRequest,
+    ) -> ProposalId {
+        let manage_neuron = create_make_proposal_payload(proposal.clone(), &neuron_id);
+        let arg = Encode!(&manage_neuron).expect("Error while encoding arg.");
+        let res = self
+            .agent
+            .update(&self.governance_principal, "manage_neuron")
+            .with_arg(arg)
+            .call_and_wait()
+            .await
+            .expect("Error while calling endpoint.");
+        //Make sure that the one making the proposal is also the controller of the neuron
+        assert_eq!(
+            PrincipalId::from(self.agent.get_principal().unwrap()),
+            neuron_controller
+        );
+        let manage_neuron_res =
+            Decode!(res.as_slice(), ManageNeuronResponse).expect("Error while decoding response.");
+        if let ManageNeuronResponse {
+            command:
+                Some(manage_neuron_response::Command::MakeProposal(MakeProposalResponse {
+                    proposal_id,
+                    ..
+                })),
+        } = manage_neuron_res
+        {
+            assert!(proposal_id.is_some());
+            let arg = Encode!(&proposal_id.unwrap().id).expect("Error while encoding arg.");
+            self.agent
+                .query(&self.governance_principal, "get_proposal_info")
+                .with_arg(arg)
+                .call()
+                .await
+                .expect("Error while calling endpoint.");
+            proposal_id.unwrap()
+        } else {
+            panic!(
+                "Making Proposal was unsuccessful --> Response : {:?}",
+                manage_neuron_res
+            )
+        }
+    }
+
+    pub async fn submit_proposal(
+        &self,
+        principal: Principal,
+        neuron_id: NeuronId,
+        title: &str,
+        summary: &str,
+        motion_text: &str,
+    ) -> ProposalId {
+        let proposal = MakeProposalRequest {
+            title: Some(title.to_string()),
+            summary: summary.to_string(),
+            action: Some(ProposalActionRequest::Motion(Motion {
+                motion_text: motion_text.to_string(),
+            })),
+            ..Default::default()
+        };
+
+        self.make_proposal(neuron_id, PrincipalId::from(principal), &proposal)
```

Comment:
What is the benefit of splitting up `submit_proposal` and `make_proposal`?

Reply:
I tend to split up functions a lot to make them shorter and easier to read, but it might be too much in this case. Fixed it.